sw/airborne/subsystems/datalink/bluegiga.c

```diff
@@ -235,7 +235,7 @@ void bluegiga_load_tx(struct bluegiga_periph *p, struct spi_transaction *trans)
     bluegiga_increment_buf(&p->tx_extract_idx, packet_len);
 
     // clear unused bytes
-    for (i = packet_len + 1; i < trans->output_length; i++) {
+    for (i = packet_len + 1; i < BLUEGIGA_BUFFER_SIZE; i++) {
```

Comment:
p->work_tx[i] has length of BLUEGIGA_BUFFER_SIZE, so you are essentially erasing the whole buffer every time (equivalent to memset)